server/sql.go

```diff
@@ -695,3 +709,107 @@ func validateSQLSelectStatement(sqlStatement string) bool {
+		Comment: fmt.Sprintf("Executed `%q` in database %q of instance %q.",
+			payload.Statement, payload.DatabaseName, payload.InstanceName),
+		Payload: string(activityBytes),
+	}
+
+	if _, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
+		log.Warn("Failed to create activity after executing sql statement",
+			zap.String("database_name", payload.DatabaseName),
+			zap.String("instance_name", payload.InstanceName),
+			zap.String("statement", payload.Statement),
+			zap.Error(err))
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create activity").SetInternal(err)
+	}
+	return nil
+}
+
+func (s *Server) sqlCheck(ctx context.Context, instance *api.Instance, exec *api.SQLExecute) (advisor.Status, []advisor.Advice, error) {
+	adviceLevel := advisor.Success
+	var adviceList []advisor.Advice
+	policy, err := s.store.GetNormalSchemaReviewPolicy(ctx, instance.EnvironmentID)
+	if err != nil {
+		if e, ok := err.(*common.Error); ok && e.Code == common.NotFound {
+			adviceLevel = advisor.Warn
+			adviceList = append(adviceList, advisor.Advice{
+				Status:  advisor.Warn,
+				Code:    common.TaskCheckEmptySchemaReviewPolicy,
+				Title:   "Empty schema review policy or disabled",
+				Content: "",
+			})
+		} else {
+			return advisor.Error, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch schema review policy by environment ID: %d", instance.EnvironmentID)).SetInternal(err)
+		}
+	}
+	if adviceLevel == advisor.Success {
+		databaseFind := &api.DatabaseFind{
+			InstanceID: &instance.ID,
+			Name:       &exec.DatabaseName,
+		}
+		db, err := s.store.FindDatabase(ctx, databaseFind)
+		if err != nil {
+			return advisor.Error, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch database `%s` for instance ID: %d", exec.DatabaseName, instance.ID)).SetInternal(err)
+		}
+		if len(db) == 0 {
+			return advisor.Error, nil, echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Database `%s` for instance ID: %d not found", exec.DatabaseName, instance.ID))
+		}
+		if len(db) > 1 {
+			return advisor.Error, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("There are multiple database `%s` for instance ID: %d", exec.DatabaseName, instance.ID))
+		}
+
+		res, err := advisor.SchemaReviewCheck(ctx, exec.Statement, policy, advisor.SchemaReviewCheckContext{
+			Charset:   db[0].CharacterSet,
+			Collation: db[0].Collation,
+			DbType:    instance.Engine,
+			Catalog:   store.NewCatalog(&db[0].ID, s.store),
+		})
+		if err != nil {
+			return advisor.Error, nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to run schema review check").SetInternal(err)
+		}
+
+		for _, advice := range res {
+			switch advice.Status {
+			case advisor.Warn:
+				if adviceLevel == advisor.Success {
+					adviceLevel = advisor.Warn
+				}
+			case advisor.Error:
+				if adviceLevel != advisor.Error {
+					adviceLevel = advisor.Error
```

Comment:
Maybe we can break the loop if we found the error level?

Reply:
I think returning all advice is better. WDYT

Reply:
My fault. The `adviceList` should filter the `Success` status, but besides, maybe we can keep the `adviceLevel` to `advisor.Error` (keep it to the worst). In the current code, the `adviceList` could contains advice with `Error` status, but the `adviceLevel` can be `Warn` instead of `Error`

Reply:
Aha, I confused myself. :(

Reply:
Fixed.